src/libfsm/exec.c

```diff
@@ -18,7 +18,7 @@
 #include "internal.h"
 
 static struct fsm_state *
-nextstate(const struct fsm_state *state, char c)
+nextstate(const struct fsm_state *state, unsigned char c)
```

Comment:
I'd rather `nextstate()` have its argument as an unqualified `char`, because that's currently what edges in FSMs are. and it's up to the caller to sort out the right if they're coming from some other type.